File: app_tieng_nhat/src/main/java/com/example/app_tieng_nhat/model/TeacherRegistrationForm.java
package com.example.app_tieng_nhat.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;
import java.util.Date;

@Getter
@Setter
@AllArgsConstructor
@RequiredArgsConstructor
@Entity
@Table(name = "teacherRegistrationForm")
public class TeacherRegistrationForm {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String teacherName;
    private String email;
    private String phone;
    private LocalDate birthDay;
    private String proof;
    private String introduce;
    private LocalDate regisDay;
    @ManyToOne
    @JsonBackReference("level-regisForm")
    @JoinColumn(name = "level_id", nullable = false)
    private Levels level;
    @ManyToOne
    @JsonBackReference("time-regisForm")
    @JoinColumn(name = "time_id", nullable = false)
    private Time workingTime;
}

File: nihonapp2_11/lib/model/TeacherRegistrationForm.dart
import 'dart:convert';

class TeacherRegistrationForm {
  final String name;
  final String email;
  final String phone;
  final String birthDay;
  final String proof; // Link ảnh Cloudinary
  final String introduce;
  final String regisDay;
  final int level_id;
  final int workingTime_id;

  TeacherRegistrationForm({
    required this.name,
    required this.email,
    required this.phone,
    required this.birthDay,
    required this.proof,
    required this.introduce,
    required this.regisDay,
    required this.level_id,
    required this.workingTime_id,
  });

  // Convert JSON to Object
  factory TeacherRegistrationForm.fromJson(Map<String, dynamic> json) {
    return TeacherRegistrationForm(
      name: json['teacherName']?.toString() ?? 'Unknown',
      email: json['email']?.toString() ?? 'No email',
      phone: json['phone']?.toString() ?? 'No phone',
      birthDay: json['birthDay']?.toString() ?? 'Unknown',
      proof: json['proof']?.toString() ?? '',
      introduce: json['introduce']?.toString() ?? 'No introduce',
      regisDay: json['regisDay']?.toString() ?? 'Unknown',
      level_id: json['level']?['id'] != null ? (json['level']['id'] as int) : 0,
      workingTime_id: json['workingTime']?['id'] != null
          ? (json['workingTime']['id'] as int)
          : 0,
    );
  }

  // Convert Object to JSON
  Map<String, dynamic> toJson() {
    return {
      "name": name,
      "email": email,
      "phone": phone,
      "birthDay": birthDay,
      "proof": proof,
      "introduce": introduce,
      "regisDay": regisDay,
      "level_id": level_id,
      "workingTime_id": workingTime_id,
    };
  }
}
